components/TextFields/src/MDCTextField.m

```diff
@@ -0,0 +1,530 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCTextFieldArt.h"
+#import "MDCTextFieldPositioningDelegate.h"
+#import "MDCTextInputCharacterCounter.h"
+#import "MDCTextInputLayoutCoordinator.h"
+
+#import "MaterialRTL.h"
+#import "MaterialTypography.h"
+
+NSString *const MDCTextFieldClearButtonColorKey = @"MDCTextFieldClearButtonColorKey";
+NSString *const MDCTextFieldClearButtonImageKey = @"MDCTextFieldClearButtonImageKey";
+NSString *const MDCTextFieldCoordinatorKey = @"MDCTextFieldCoordinatorKey";
+NSString *const MDCTextFieldTextDidSetTextNotification = @"MDCTextFieldTextDidSetTextNotification";
+
+static const CGFloat MDCTextInputTextRectRightPaddingCorrection = -4.f;
```

Comment:
(Any similar comments may be ignored if we are specifically talking about the leftView or rightView.